alica_dummy_proxy/CMakeLists.txt

```diff
@@ -35,7 +35,13 @@ include_directories(
 file(GLOB_RECURSE alica_dummy_sources "src/*.cpp")
 add_library(alica_dummy_proxy ${alica_dummy_sources})
 target_link_libraries(alica_dummy_proxy ${catkin_LIBRARIES})
- 
+
 if (NOT catkin_FOUND)
- target_include_directories(alica_dummy_proxy PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include") 
+ target_include_directories(alica_dummy_proxy PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
 endif (NOT catkin_FOUND)
+
+install(TARGETS ${PROJECT_NAME}
```

Comment:
Do you also need to install the header files? Like this:

`install(DIRECTORY include/
  DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION}
  FILES_MATCHING PATTERN "*.h" PATTERN "*.hpp"
  PATTERN ".svn" EXCLUDE
)`

Example taken from [CMakeLists.txt of AlicaEngine](https://github.com/rapyuta-robotics/alica/blob/rr-devel/alica_engine/CMakeLists.txt#L57)

Reply:
I believe catkin did not complain because tests are not installed, and the headers are only used in the tests of `alica_tests`.  However, even without installing tests, it fails in `find_package()` if you do not do `install(TARGETS ...`

Either way it is fine to install the headers whiel we are chaning things anyway

Reply:
> install(DIRECTORY include/ DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION} FILES_MATCHING PATTERN "*.h" PATTERN "*.hpp" PATTERN ".svn" EXCLUDE )
Is there a reason for `CATKIN_GLOBAL_INCLUDE_DESTINATION` over `CATKIN_PACKAGE_INCLUDE_DESTINATION`?

Reply:
I don't know about any preferences about this in general. Maybe @abhi1287 can comment on this.

Reply:
As far as I can tell both seem to be used without much thought into it, so I guess it probably doesn't matter.  In a PR to alica-supplementary the part I copy-pasted to install the includes ended up using the global destination.

Reply:
CATKIN_PACKAGE_INCLUDE_DESTINATION  is always preferred. For components that use CATKIN_GLOBAL_INCLUDE_DESTINATION  we had to do it because of a legacy issue. Headers were not placed in   "include/<package_name>/"  but rather "include/"  and included all over in applications using #include"header.h" format rather than #include "package/header.h" .  To avoid breaking them, we decided to copy headers to both package as well as global destination.

Reply:
@abhi1287 Got it.  Here I used CATKIN_PACKAGE_INCLUDE_DESTINATION so it should be okay.  I will change the other two PRs and test